sync/sync.go

```diff
@@ -222,17 +222,17 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				return
 			}
 			highestBlockHeader := s.highestBlockHeader.Load()
-			if highestBlockHeader == nil || highestBlockHeader.Number <= block.Number {
-				highestBlock, err := s.starknetData.BlockLatest(ctx)
-				if err != nil {
-					s.log.Warnw("Failed fetching latest block", "err", err)
-				} else {
-					s.highestBlockHeader.Store(highestBlock.Header)
-					isBehind := highestBlock.Number > block.Number+uint64(maxWorkers())
-					if s.catchUpMode != isBehind {
-						resetStreams()
-					}
-					s.catchUpMode = isBehind
+			if highestBlockHeader != nil {
+				isBehind := highestBlockHeader.Number > block.Number+uint64(maxWorkers())
+				if s.catchUpMode != isBehind {
+					resetStreams()
+				}
```

Comment:
If `s.catchUpMode == true` and `isBehind == false` then we would reset the streams. Is that the expected behaviour? Shouldn't we only reset the stream when `s.catchUpMode == false` and `isBehind == true`? I don't think the `!=` operator is the correct here.

Reply:
We want to reset in both cases. If `catchupMode != isBehind`, it means that we either fell behind and need to enable catchup mode or managed to catch up to HEAD and need to disable it. We need to reset the sync loop when both enabling and disabling catchup mode.